File: system-test/typescript/src/testkit/dsl/port/when/steps/when-publish-coupon.ts
import type { ThenResultStage } from '../../then/then-result-stage.js';

export interface WhenPublishCoupon {
  withCouponCode(couponCode: string): this;
  withDiscountRate(discountRate: string | number): this;
  withValidFrom(validFrom: string | null): this;
  withValidTo(validTo: string | null): this;
  withUsageLimit(usageLimit: string | number | null): this;
  then(): ThenResultStage;
}

File: system/multitier/backend-java/src/testSupport/java/com/mycompany/myshop/backend/support/port/when/steps/WhenPublishCoupon.java
package com.mycompany.myshop.backend.support.port.when.steps;

import com.mycompany.myshop.backend.support.port.when.steps.base.WhenStep;

public interface WhenPublishCoupon extends WhenStep {
    WhenPublishCoupon withCouponCode(String couponCode);

    WhenPublishCoupon withDiscountRate(String discountRate);

    WhenPublishCoupon withDiscountRate(double discountRate);

    WhenPublishCoupon withValidFrom(String validFrom);

    WhenPublishCoupon withValidTo(String validTo);

    WhenPublishCoupon withUsageLimit(int usageLimit);
}
